src/app/ui-library/callout.tsx

```diff
@@ -0,0 +1,108 @@
+import { ArrowRight, CheckCircle, Info, Warning, WarningCircle } from '@phosphor-icons/react';
+import { createElement } from 'react';
+import styled from 'styled-components';
+import Theme from 'theme';
+import { StyledHeading, StyledP } from './common.styled';
+
+type CalloutVariant = 'info' | 'warning' | 'danger' | 'success';
+
+const backgroundColors = {
+  info: Theme.colors.elevation6_600,
+  warning: Theme.colors.caution_800,
+  success: Theme.colors.success_dark_600,
+  danger: Theme.colors.danger_dark_600,
+};
+const getBackgroundForVariant = (variant: CalloutVariant) => backgroundColors[variant];
+
+const icons = {
+  info: Info,
+  warning: WarningCircle,
+  success: CheckCircle,
+  danger: Warning,
+};
+const getIconForVariant = (variant: CalloutVariant) =>
+  createElement(icons[variant], {
+    weight: 'fill',
+    size: '24',
+    color: Theme.colors.white_200,
+    style: {
+      flexShrink: 0,
+      flexGrow: 0,
+    },
+  });
```

Comment:
nite: Would prefer to consistently use `styled` and keep away from lower level react functions

```suggestion
const getIconForVariant = (variant: CalloutVariant) => styled(icons[variant])`
  flex-shrink: 0;
  flex-grow: 0;
  color: ${Theme.colors.white_200};
`;
```

and in the render:
```
  const Icon = getIconForVariant(variant);
  return (
    <Container variant={variant}>
      <Icon size={24} weight="fill" />
```

Reply:
yeah true. I think styled-components has a way to pass attributes, which was what I was originally using createElement for. I'll look into this

Reply:
You can use the code in my comment above. It's tested and working 😋

Reply:
get some weird render behaviour and this warning with that approach. needs something different: 
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/011ae483-9d03-4ada-ac1b-c0dadbb7a09f)

Reply:
Ah, ok. I guess it creates a new one every time 😅

Reply:
This works:
```
const getStyledIconForVariant = (icon: Icon) =>
  styled(icon)`
    color: ${(props) => props.theme.colors.white_200},
    flex-shrink: 0,
    flex-grow: 0,
`;

const icons = {
  info: getStyledIconForVariant(Info),
  warning: getStyledIconForVariant(WarningCircle),
  success: getStyledIconForVariant(CheckCircle),
  danger: getStyledIconForVariant(Warning),
};
```

and in render:
```
const VariantIcon = icons[variant];
  return (
    <Container variant={variant}>
      <VariantIcon size={24} weight="fill" />
      <TextContainer>
```